File: src/main/java/com/mssoftech/dbflute/bsentity/dbmeta/ServiceRankDbm.java
package com.mssoftech.dbflute.bsentity.dbmeta;

import java.util.List;
import java.util.Map;

import org.dbflute.Entity;
import org.dbflute.dbmeta.AbstractDBMeta;
import org.dbflute.dbmeta.info.*;
import org.dbflute.dbmeta.name.*;
import org.dbflute.dbmeta.property.PropertyGateway;
import org.dbflute.dbway.DBDef;
import com.mssoftech.dbflute.allcommon.*;
import com.mssoftech.dbflute.exentity.*;

/**
 * The DB meta of service_rank. (Singleton)
 * @author DBFlute(AutoGenerator)
 */
public class ServiceRankDbm extends AbstractDBMeta {

    // ===================================================================================
    //                                                                           Singleton
    //                                                                           =========
    private static final ServiceRankDbm _instance = new ServiceRankDbm();
    private ServiceRankDbm() {}
    public static ServiceRankDbm getInstance() { return _instance; }

    // ===================================================================================
    //                                                                       Current DBDef
    //                                                                       =============
    public String getProjectName() { return DBCurrent.getInstance().projectName(); }
    public String getProjectPrefix() { return DBCurrent.getInstance().projectPrefix(); }
    public String getGenerationGapBasePrefix() { return DBCurrent.getInstance().generationGapBasePrefix(); }
    public DBDef getCurrentDBDef() { return DBCurrent.getInstance().currentDBDef(); }

    // ===================================================================================
    //                                                                    Property Gateway
    //                                                                    ================
    // -----------------------------------------------------
    //                                       Column Property
    //                                       ---------------
    protected final Map<String, PropertyGateway> _epgMap = newHashMap();
    { xsetupEpg(); }
    protected void xsetupEpg() {
        setupEpg(_epgMap, et -> ((ServiceRank)et).getServiceRankCode(), (et, vl) -> ((ServiceRank)et).setServiceRankCode((String)vl), "serviceRankCode");
        setupEpg(_epgMap, et -> ((ServiceRank)et).getServiceRankName(), (et, vl) -> ((ServiceRank)et).setServiceRankName((String)vl), "serviceRankName");
        setupEpg(_epgMap, et -> ((ServiceRank)et).getServicePointIncidence(), (et, vl) -> ((ServiceRank)et).setServicePointIncidence(ctb(vl)), "servicePointIncidence");
        setupEpg(_epgMap, et -> ((ServiceRank)et).getNewAcceptableFlg(), (et, vl) -> ((ServiceRank)et).setNewAcceptableFlg(cti(vl)), "newAcceptableFlg");
        setupEpg(_epgMap, et -> ((ServiceRank)et).getDescription(), (et, vl) -> ((ServiceRank)et).setDescription((String)vl), "description");
        setupEpg(_epgMap, et -> ((ServiceRank)et).getDisplayOrder(), (et, vl) -> ((ServiceRank)et).setDisplayOrder(cti(vl)), "displayOrder");
    }
    public PropertyGateway findPropertyGateway(String prop)
    { return doFindEpg(_epgMap, prop); }

    // ===================================================================================
    //                                                                          Table Info
    //                                                                          ==========
    protected final String _tableDbName = "service_rank";
    protected final String _tableDispName = "service_rank";
    protected final String _tablePropertyName = "serviceRank";
    protected final TableSqlName _tableSqlName = new TableSqlName("service_rank", _tableDbName);
    { _tableSqlName.xacceptFilter(DBFluteConfig.getInstance().getTableSqlNameFilter()); }
    public String getTableDbName() { return _tableDbName; }
    public String getTableDispName() { return _tableDispName; }
    public String getTablePropertyName() { return _tablePropertyName; }
    public TableSqlName getTableSqlName() { return _tableSqlName; }

    // ===================================================================================
    //                                                                         Column Info
    //                                                                         ===========
    protected final ColumnInfo _columnServiceRankCode = cci("service_rank_code", "service_rank_code", null, null, String.class, "serviceRankCode", null, true, false, true, "bpchar", 3, 0, null, false, null, null, null, "memberServiceList", null, false);
//"string"
    protected final ColumnInfo _columnServiceRankName = cci("service_rank_name", "service_rank_name", null, null, String.class, "serviceRankName", null, false, false, true, "varchar", 50, 0, null, false, null, null, null, null, null, false);
//"string"
    protected final ColumnInfo _columnServicePointIncidence = cci("service_point_incidence", "service_point_incidence", null, null, java.math.BigDecimal.class, "servicePointIncidence", null, false, false, true, "numeric", 5, 3, null, false, null, null, null, null, null, false);
//"df.Numeric"
    protected final ColumnInfo _columnNewAcceptableFlg = cci("new_acceptable_flg", "new_acceptable_flg", null, null, Integer.class, "newAcceptableFlg", null, false, false, true, "int4", 10, 0, null, false, null, null, null, null, null, false);
//"int64"
    protected final ColumnInfo _columnDescription = cci("description", "description", null, null, String.class, "description", null, false, false, true, "varchar", 200, 0, null, false, null, null, null, null, null, false);
//"string"
    protected final ColumnInfo _columnDisplayOrder = cci("display_order", "display_order", null, null, Integer.class, "displayOrder", null, false, false, true, "int4", 10, 0, null, false, null, null, null, null, null, false);
//"int64"

    /**
     * service_rank_code: {PK, NotNull, bpchar(3)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnServiceRankCode() { return _columnServiceRankCode; }
    /**
     * service_rank_name: {NotNull, varchar(50)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnServiceRankName() { return _columnServiceRankName; }
    /**
     * service_point_incidence: {NotNull, numeric(5, 3)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnServicePointIncidence() { return _columnServicePointIncidence; }
    /**
     * new_acceptable_flg: {NotNull, int4(10)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnNewAcceptableFlg() { return _columnNewAcceptableFlg; }
    /**
     * description: {NotNull, varchar(200)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnDescription() { return _columnDescription; }
    /**
     * display_order: {UQ, NotNull, int4(10)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnDisplayOrder() { return _columnDisplayOrder; }

    protected List<ColumnInfo> ccil() {
        List<ColumnInfo> ls = newArrayList();
        ls.add(columnServiceRankCode());
        ls.add(columnServiceRankName());
        ls.add(columnServicePointIncidence());
        ls.add(columnNewAcceptableFlg());
        ls.add(columnDescription());
        ls.add(columnDisplayOrder());
        return ls;
    }

    { initializeInformationResource(); }

    // ===================================================================================
    //                                                                         Unique Info
    //                                                                         ===========
    // -----------------------------------------------------
    //                                       Primary Element
    //                                       ---------------
    protected UniqueInfo cpui() { return hpcpui(columnServiceRankCode()); }
    public boolean hasPrimaryKey() { return true; }
    public boolean hasCompoundPrimaryKey() { return false; }

    // -----------------------------------------------------
    //                                        Unique Element
    //                                        --------------
    public UniqueInfo uniqueOf() { return hpcui(columnDisplayOrder()); }

    // ===================================================================================
    //                                                                       Relation Info
    //                                                                       =============
    // cannot cache because it uses related DB meta instance while booting
    // (instead, cached by super's collection)
    // -----------------------------------------------------
    //                                      Foreign Property
    //                                      ----------------

    // -----------------------------------------------------
    //                                     Referrer Property
    //                                     -----------------
    /**
     * member_service by service_rank_code, named 'memberServiceList'.
     * @return The information object of referrer property. (NotNull)
     */
    public ReferrerInfo referrerMemberServiceList() {
        Map<ColumnInfo, ColumnInfo> mp = newLinkedHashMap(columnServiceRankCode(), MemberServiceDbm.getInstance().columnServiceRankCode());
        return cri("fk_member_service_service_rank_code", "memberServiceList", this, MemberServiceDbm.getInstance(), mp, false, "serviceRank");
    }

    // ===================================================================================
    //                                                                        Various Info
    //                                                                        ============

    // ===================================================================================
    //                                                                           Type Name
    //                                                                           =========
    public String getEntityTypeName() { return "com.mssoftech.dbflute.exentity.ServiceRank"; }
    public String getConditionBeanTypeName() { return "com.mssoftech.dbflute.cbean.ServiceRankCB"; }
    public String getBehaviorTypeName() { return "com.mssoftech.dbflute.exbhv.ServiceRankBhv"; }

    // ===================================================================================
    //                                                                         Object Type
    //                                                                         ===========
    public Class<ServiceRank> getEntityType() { return ServiceRank.class; }

    // ===================================================================================
    //                                                                     Object Instance
    //                                                                     ===============
    public ServiceRank newEntity() { return new ServiceRank(); }

    // ===================================================================================
    //                                                                   Map Communication
    //                                                                   =================
    public void acceptPrimaryKeyMap(Entity et, Map<String, ? extends Object> mp)
    { doAcceptPrimaryKeyMap((ServiceRank)et, mp); }
    public void acceptAllColumnMap(Entity et, Map<String, ? extends Object> mp)
    { doAcceptAllColumnMap((ServiceRank)et, mp); }
    public Map<String, Object> extractPrimaryKeyMap(Entity et) { return doExtractPrimaryKeyMap(et); }
    public Map<String, Object> extractAllColumnMap(Entity et) { return doExtractAllColumnMap(et); }
}

File: src/dbflute/adf/meta/serviceRankdbm.go
package meta

import (
	"github.com/mikeshimura/dbflute/df"
)

type ServiceRankDbm_T struct {
	df.BaseDBMeta
	ColumnServiceRankCode *df.ColumnInfo
	ColumnServiceRankName *df.ColumnInfo
	ColumnServicePointIncidence *df.ColumnInfo
	ColumnNewAcceptableFlg *df.ColumnInfo
	ColumnDescription *df.ColumnInfo
	ColumnDisplayOrder *df.ColumnInfo
}

func (b *ServiceRankDbm_T) GetProjectName() string {
	return df.DBCurrent_I.ProjectName
}

func (b *ServiceRankDbm_T) GetDbCurrent() *df.DBCurrent {
	return df.DBCurrent_I
}

var ServiceRankDbm *ServiceRankDbm_T

func Create_ServiceRankDbm() {
	ServiceRankDbm = new(ServiceRankDbm_T)
	ServiceRankDbm.TableDbName = "service_rank"
	ServiceRankDbm.TableDispName = "service_rank"
	ServiceRankDbm.TablePropertyName = "serviceRank"
	ServiceRankDbm.TableSqlName = new(df.TableSqlName)
	ServiceRankDbm.TableSqlName.TableSqlName = "service_rank"
	ServiceRankDbm.TableSqlName.CorrespondingDbName = ServiceRankDbm.TableDbName

	var serviceRank df.DBMeta
	serviceRank = ServiceRankDbm
	ServiceRankDbm.DBMeta=&serviceRank
	serviceRankCodeSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo service_rank_code
	serviceRankCodeSqlName.ColumnSqlName = "service_rank_code"
	serviceRankCodeSqlName.IrregularChar = false
	ServiceRankDbm.ColumnServiceRankCode = df.CCI(&serviceRank, "service_rank_code", serviceRankCodeSqlName, "", "", "String.class", "serviceRankCode", "", true, false,true, "bpchar", 3, 0, "",false,"","", "","memberServiceList","",false,"string")
	serviceRankNameSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo service_rank_name
	serviceRankNameSqlName.ColumnSqlName = "service_rank_name"
	serviceRankNameSqlName.IrregularChar = false
	ServiceRankDbm.ColumnServiceRankName = df.CCI(&serviceRank, "service_rank_name", serviceRankNameSqlName, "", "", "String.class", "serviceRankName", "", false, false,true, "varchar", 50, 0, "",false,"","", "","","",false,"string")
	servicePointIncidenceSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo service_point_incidence
	servicePointIncidenceSqlName.ColumnSqlName = "service_point_incidence"
	servicePointIncidenceSqlName.IrregularChar = false
	ServiceRankDbm.ColumnServicePointIncidence = df.CCI(&serviceRank, "service_point_incidence", servicePointIncidenceSqlName, "", "", "java.math.BigDecimal.class", "servicePointIncidence", "", false, false,true, "numeric", 5, 3, "",false,"","", "","","",false,"df.Numeric")
	newAcceptableFlgSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo new_acceptable_flg
	newAcceptableFlgSqlName.ColumnSqlName = "new_acceptable_flg"
	newAcceptableFlgSqlName.IrregularChar = false
	ServiceRankDbm.ColumnNewAcceptableFlg = df.CCI(&serviceRank, "new_acceptable_flg", newAcceptableFlgSqlName, "", "", "Integer.class", "newAcceptableFlg", "", false, false,true, "int4", 10, 0, "",false,"","", "","","",false,"int64")
	descriptionSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo description
	descriptionSqlName.ColumnSqlName = "description"
	descriptionSqlName.IrregularChar = false
	ServiceRankDbm.ColumnDescription = df.CCI(&serviceRank, "description", descriptionSqlName, "", "", "String.class", "description", "", false, false,true, "varchar", 200, 0, "",false,"","", "","","",false,"string")
	displayOrderSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo display_order
	displayOrderSqlName.ColumnSqlName = "display_order"
	displayOrderSqlName.IrregularChar = false
	ServiceRankDbm.ColumnDisplayOrder = df.CCI(&serviceRank, "display_order", displayOrderSqlName, "", "", "Integer.class", "displayOrder", "", false, false,true, "int4", 10, 0, "",false,"","", "","","",false,"int64")

	ServiceRankDbm.ColumnInfoList = new(df.List)
	ServiceRankDbm.ColumnInfoList.Add(ServiceRankDbm.ColumnServiceRankCode)
	ServiceRankDbm.ColumnInfoList.Add(ServiceRankDbm.ColumnServiceRankName)
	ServiceRankDbm.ColumnInfoList.Add(ServiceRankDbm.ColumnServicePointIncidence)
	ServiceRankDbm.ColumnInfoList.Add(ServiceRankDbm.ColumnNewAcceptableFlg)
	ServiceRankDbm.ColumnInfoList.Add(ServiceRankDbm.ColumnDescription)
	ServiceRankDbm.ColumnInfoList.Add(ServiceRankDbm.ColumnDisplayOrder)


	ServiceRankDbm.ColumnInfoMap=make(map[string]int)
	ServiceRankDbm.ColumnInfoMap["serviceRankCode"]=0
		ServiceRankDbm.ColumnInfoMap["serviceRankName"]=1
		ServiceRankDbm.ColumnInfoMap["servicePointIncidence"]=2
		ServiceRankDbm.ColumnInfoMap["newAcceptableFlg"]=3
		ServiceRankDbm.ColumnInfoMap["description"]=4
		ServiceRankDbm.ColumnInfoMap["displayOrder"]=5
	    ServiceRankDbm.PrimaryKey = true
    ServiceRankDbm.CompoundPrimaryKey = false
	ui := new(df.UniqueInfo)
	ui.DbMeta = &serviceRank
	ui.Primary = true
	ui.UniqueColumnList = new(df.List)
	ui.UniqueColumnList.Add(ServiceRankDbm.ColumnServiceRankCode)

	ServiceRankDbm.PrimaryInfo = new(df.PrimaryInfo)
	ServiceRankDbm.PrimaryInfo.UniqueInfo = ui
	
	var serviceRankMeta df.DBMeta = ServiceRankDbm
	df.DBMetaInstanceHandler_I.TableDbNameInstanceMap["ServiceRank"] = &serviceRankMeta
}
